### PR TITLE
Created Setting Builder Activity

### PR DESCRIPTION
**Pull request by cloudthepain**

### CODE REVIEW

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 1]
```diff
@@ -0,0 +1,150 @@
+@inherits CaptainCoder.Page
+@page "/activities/setting-builder"
+
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:3]: Add a <Title>Setting Builder</Title>

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+<Section Title="Setting Builder">
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:4]: First section of each page should be called "Introduction" and include an "Adventure / AdventureDialog". You should put  your Avatar here and introduce yourself. Or, maybe add in a character who is a "City Planner"?

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    <Text>
+        When we thinking about making a game a lot of things can come to mind. From
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:6]: Avoid using "we" when possible. Maybe something like, "Many things can come to mind when thinking about creating a game."

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    <MudList>
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Fantasy</MudListItem>
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Horror</MudListItem>
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Adventure</MudListItem>
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Sci-fi</MudListItem>
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Post-Apocalyptic</MudListItem>
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Mystery</MudListItem>
+    </MudList>
+</Section>
+    <TaskItem Name="Create Project"><Code>Create a project called "Setting Builder"</Code></TaskItem>
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:58]: No need to use <Code> blocks on task items. Also, consider using the <ConsoleProjectTask> which puts a button that allows them to see a cheat sheet for VS Code for setting up a project.

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    <TaskItem Name="Create Introduction"><Code>Create an introduction to tell the user what your program is.</Code></TaskItem>
+    <TaskItem Name="Ask for Genre"><Code>Ask the user for a genre input and then Display a list of genres to help them out.</Code></TaskItem>
+    <TaskItem Name="Store user Input"><Code>Store the users input as "genreInput".</Code></TaskItem>
+
+    <Text>
+        After you've completed the assignment above you should have a result that looks something like this. Don't worry
+        if yours doesn't match the example exactly.
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:65]: Did you mean to have an <Output> here or a ReplIt?

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    <TaskItem Name="Store natural resources"><Code>Store the natural resources as "naturalResources".</Code></TaskItem>
+
+    <Section Title="Displaying the information">
+    <Text>
+        Wow! We've got a lot of information going now! I think it is about time to start looking at how to Display the information
+        in a way that looks good for our user. While we could just Display the information out as "information: ", "user input" format,
+        it might be a little hard to read the information. Especially if we want to add more information.
+    </Text>
+    <Text>
+        What we should do instead is change the color of the information in each row to make it a little easier to read!
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:120]: Maybe instead of "we should", maybe something like: "You can make your program easier to read by highlighting the information with colors."

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor (hunk 2 above)
Comment (jcollard) [reply IntroToCSharp/Pages/Activities/SettingBuilder.razor:4]: They might be a character saying, "It sure is tough to plan a city! I can never remember all of the things that are involved. It would be so helpful if you could create a program to help me remember the steps."

**Review by jcollard — CHANGES_REQUESTED** [review]
A great start! I've left some feedback inline.

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor (hunk 4 above)
Comment (cloudthepain) [reply IntroToCSharp/Pages/Activities/SettingBuilder.razor:58]: Sorry, could you rephrase this? I don't understand what you are saying.

Comment (jcollard) [reply IntroToCSharp/Pages/Activities/SettingBuilder.razor:58]: You don't need the code blocks. Can be re-written as:
```
<TaskItem Name="Some Name">Task Goes here</TaskItem>
```

Also, there is a special component for doing "Create Project" tasks:
```
<ConsoleProjectTask Name="Create Project>Create a project called "Stting Builder"</ConsoleProjectTask>
```

**Comment by cloudthepain** [comment]
Finished editing. Would like a review of new edits.
@jcollard

**Comment by jcollard** [comment]
@cloudthepain Looks like you have some merge conflict. Make sure to `get merge origin/develop` into this branch to resolve them.

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+                <Text>
+                    In the following activity you are going to create a program that will help me plan
+                    out a city or town. You are going to need to prompt the user for information
+                    and be able to save that information in a variable using <Code>Console.WriteLine</Code>,
+                    <Code>Console.ReadLine</Code>, and the basics of <Code>variables</Code>.
+                </Text>
+            </AdventureDialog>
+        </Item>
+        <Item>
+            <AdventureDialog>
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:48]: Use the `Page` field in `AdventureDialog` to add in a "smart" button which will allow the user to add the pages if they are missing. You will have to break this into two Dialogs, one for each.

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+        When you think of many of your favorite games what comes to mind? Perhaps you think of an astronaut
+        flying through an astreroid field in space like in No Man's Sky. Or maybe you think of a plumber jumping
+        on evil turtles to Store a princess like Mario. Maybe it is even something scarier where you are trapped
+        in a Kid's Pizza store with animatronics trying to come to life and get you!
+    </Text>
+    <Text>
+        There are so many different settings that games or stories might take place but the first thing you should figure out
+        is what kind of "genre" the setting is in. Genre is a term that generally means the particular "style" and
+        "world" of the game or story. By knowing your genre you know what places <i>could</i> be possible and what places 
+        wouldn't be possible. you wouldn't expect to see a space level in a fantasy game like Skyrim!
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:79]: `For example, you wouldn't...`

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+    </Text>
+    <Text>
+        To help the user figure out what genre the setting is in, let's give our users a list of options to help them
+        generate their own ideas for a genre. You can use the following list as a base but feel free to add more if you want!
+    </Text>
+    <Note>
+        <b>Extra Challenge!</b><br>
+        Can you figure out how to add bullet points to your list as youll? Hint: You should check out the <b>@AsciiArt.PageRef.Name</b> lesson
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:87]: Double check sentence: "Can you figure out how to add bullet points to your list as youll?"

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Horror</MudListItem>
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Adventure</MudListItem>
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Sci-fi</MudListItem>
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Post-Apocalyptic</MudListItem>
+        <MudListItem Avatar="@Icons.Filled.MenuBook">Mystery</MudListItem>
+    </MudList>
+</Section>
+    <TaskItem Name="Create Project">Create a project called "Setting Builder"</TaskItem>
+    <TaskItem Name="Create Introduction">Create an introduction to tell the user what your program is.</TaskItem>
+    <TaskItem Name="Ask for Genre">Ask the user for a genre input and then Display a list of genres to help them out.</TaskItem>
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:101]: Break into two tasks: "Display a list of genres" and "Prompt the user to enter a genre (string)."

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+    <TaskItem Name="Store user Input">Store the users input as "genreInput".</TaskItem>
+
+<Section Title="Somewhere to call home.">
+    <Text>
+        Great job! You have a start to your setting builder! Now you can get into some fun by building your first in your 
+        Setting Builder. We'll need to think about the details of our program as if we were building our own setting.
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:107]: Proof read and revise: "Now you can get into some fun by building your first in your Setting Builder."

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+        When we think of a city what is the first thing that comes to mind? For me, it is how big of a city is it?
+        Is it filled with skyscrapes that cut the clouds or is it perhaps closer to a small town with farms full of crops ready to be 
+        harvested as far as the eye can see?
+    </Text>
+    <Text>
+        To get the user going for building their first city or town let's ask them how big their city or town is going to be.
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:113]: Try to be a little more formal with your writing. Rather than, "To get the user going" use "Next, ask the user how big their city or town will be."

File: IntroToCSharp/Pages/Activities/SettingBuilder.razor [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+    <TaskItem Name="Alternate Colors">Alternate the colors displayed on each line to the user to make it easier to read.</TaskItem>
+
+    </Section>
+    <Section Title="Wrapping Up">
+    <Text>
+        By the end of this project your Display out should look something like this!
+    </Text>
+    <CodeBlock ReplIt="NobodyOdyssues/SettingBuilder"> City Builder</CodeBlock>
+    </Section>
+    <Section Title="Wrapping Up">
```
Comment (jcollard) [inline IntroToCSharp/Pages/Activities/SettingBuilder.razor:167]: Rename section to "What's Next?" and include a link to at least one other section that is semi-related or would be a good place to go after this.

**Review by jcollard — CHANGES_REQUESTED** [review]
In general, I would like for the activities to be a little more formal.

Checkout the comments I've added in then go back and do a full re-read of the activity making changes as you go.

**Comment by cloudthepain** [comment]
@jcollard 
I believe I have handled all the merge errors and it is ready to be merged into develop.
